.github/workflows/reusable_checks_cpp.yml

```diff
@@ -0,0 +1,85 @@
+name: "C++ Tests on all platforms & compilers"
+
+on:
+  workflow_call:
+    inputs:
+      CONCURRENCY:
+        required: true
+        type: string
+      MINIMAL:
+        required: false
+        type: boolean
+        default: false
+
+  workflow_dispatch:
+    inputs:
+      CONCURRENCY:
+        required: false
+        type: string
+        default: "adhoc"
+      MINIMAL:
+        type: boolean
+        default: false
+        description: "If true only linux is tested"
+
+concurrency:
+  group: ${{ inputs.CONCURRENCY }}-checks-cpp
+  cancel-in-progress: true
+
+env:
+  # See: https://github.com/marketplace/actions/sccache-action
+  SCCACHE_GHA_ENABLED: "false"
+
+permissions:
+  contents: "read"
+  id-token: "write"
+
+jobs:
+  matrix_prep:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.ref || '' }}
+      - name: Load minimal C++ test matrix
+        if: ${{ github.event.inputs.MINIMAL }}
+        run: echo "matrix=$(jq -c . < ./.github/workflows/cpp_matrix_full.json)" >> $GITHUB_OUTPUT
+      - name: Load full C++ test matrix
+        if: ${{ !github.event.inputs.MINIMAL }}
+        run: echo "matrix=$(jq -c . < ./.github/workflows/cpp_matrix_minimal.json)" >> $GITHUB_OUTPUT
```

Comment:
It is good practice to use proper antonyms for opposites, and I'm not sure "full" and "minimal" qualifies. Maybe:

* partial / full
* minimal / maximal
* some / all
* fast / slow